examples/auth-flow-async-with-query-params/app.js

```diff
@@ -0,0 +1,97 @@
+import React, { createClass, PropTypes } from 'react'
+import { render } from 'react-dom'
+import { Router, Route, IndexRoute, browserHistory, Link } from 'react-router'
+
+function App(props) {
+  return (
+    <div>
+      {props.children}
+    </div>
+  )
+}
+
+const Form = createClass({
+  contextTypes: {
+    router: PropTypes.object.isRequired
+  },
+
+  getInitialState() {
+    return {
+      value: ''
+    }
+  },
+
+  submitAction(event) {
+    event.preventDefault()
+    this.context.router.push({
+      pathname: '/page',
+      query: {
+        qsparam: this.state.value
+      }
+    })
+  },
+
+  handleChange(event) {
+    this.setState({ value: event.target.value })
+  },
+
+  render() {
+    return (
+      <form onSubmit={this.submitAction}>
+        <p>Token is <em>pancakes</em></p>
+        <input type="text" value={this.state.value} onChange={this.handleChange} />
+        <button type="submit">Submit the thing</button>
+        <p><Link to="/page?qsparam=pancakes">Or authenticate via URL</Link></p>
+        <p><Link to="/page?qsparam=bacon">Or try failing to authenticate via URL</Link></p>
+      </form>
+    )
+  }
+})
+
+function Page() {
+  return <h1>Hey I see you've authenticated. Welcome!</h1>
```

Comment:
Nah, it's just some funkiness with [Linguist](https://github.com/github/linguist) and its parsing of JSX within Javascript.